ledger/metrics/src/main/scala/com/daml/metrics/MetricHandle.scala

```diff
@@ -24,111 +21,166 @@ object MetricHandle {
 
     def prefix: MetricName
 
-    def registry: codahale.MetricRegistry
-
-    def timer(name: MetricName): Timer = Timer(name, registry.timer(name))
+    def timer(name: MetricName): Timer
 
-    def varGauge[T](name: MetricName, initial: T): VarGauge[T] =
-      addGauge(name, Gauges.VarGauge[T](initial), _.updateValue(initial))
+    def gauge[T](name: MetricName, initial: T): Gauge[T]
 
-    def addGauge[T <: codahale.Gauge[M], M](
+    def gaugeWithSupplier[T](
```

Comment:
removing `addGauge` will break canton in 
https://github.com/DACH-NY/canton/blob/e888db5dc5ab7dccba61ebef171fb2e5c46b4d3d/community/common/src/main/scala/com/digitalasset/canton/metrics/MetricHandle.scala#L30-L36

Reply:
Based on what I'm seeing in canton, the `RefGauge` is more or less a `VarGauge`. The metrics using `RefGauge` should in fact use a `gaugeWithSupplier`, and the `RefGuage` should be deleted.